frontend/talentsearch/src/js/components/experienceForm/ExperienceForm.tsx

```diff
@@ -140,8 +132,43 @@ export const ExperienceForm: React.FunctionComponent<ExperienceFormProps> = ({
       },
       ...crumbs,
     ];
+
+    if (experience) {
+      irrelevantSkills = experience?.skills?.filter((skill) => {
+        return (
+          !poolAdvertisement.essentialSkills?.find(
+            (essentialSkill) => essentialSkill.id === skill.id,
+          ) &&
+          !poolAdvertisement.nonessentialSkills?.find(
+            (assetSkill) => assetSkill.id === skill.id,
+          )
+        );
+      });
+      // eslint-disable-next-line no-param-reassign
+      experience.skills = experience?.skills?.filter(
+        (skill) =>
+          !irrelevantSkills?.find(
+            (irrelevantSkill) => irrelevantSkill.id === skill.id,
+          ),
+      );
```

Comment:
Not sure if it counts as within scope of this issue but we don't really need to ignore this rule since it is recommend to use use a differnt config than we are currently using.

**Recommended**

```json
{
  "no-param-reassign": ["error", { "props": false }]
}
```

**Current**

```json
{
  "no-param-reassign": "warn"
}
```

REF: https://stackoverflow.com/a/42399879

Another option would to just use a new variable instead of modifying experience directly.

```tsx
const newExperience = {...experience};
newExperience.skills = ...;
```

Reply:
I've run into this one before and just created a new variable like you suggested.  The linting rule change seems reasonable but  I'm not sure changing it would fix this case.  It's not a prop of skills that is being changed, it's skills itself, right?

Reply:
I actually tested it and it does make this pass without ignoring the rule. However, I am not opposed to leaving the rule alone and just creating a new variable.

Reply:
@petertgiles skills is a prop of experience, so I think the linting change would work.

However, even if it avoids the arg bugs described in that stackoverflow answer, I would avoid this kind of reassignment on principle 😅. Especially modifying something that was passed into the function/component as a prop, which is a case of [modifying something outside of this function's scope](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/documentation/style-guide.md#main-principles).

I'd much prefer `const skills = experience?.skills?.filter(...);`, or if you really need them in the full package,
```
const newExperience = {
  ...experience,
  skills = ...,
};
```